### PR TITLE
refactor / dockerfile to reduce image size

### PR DESCRIPTION
**Pull request by bitfag**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue:
- Related Clubhouse Story: https://app.clubhouse.io/coinalpha/story/9209/reduce-docker-image-size-for-hummingbot

**A description of the changes proposed in the pull request**:

Significantly reduced image size. Regarding MKL removal, I changed dependencies in setup/environment-linux.yml only as I'm unsure how to get proper versions for mac/windows. Also note that numpy version slightly changed. After building an image, I tested that hummingbot could be launched using this image.

### CODE REVIEW

**Comment by rennel-tabing** [comment]
I am getting 2.24 GB in size for this docker build

**Comment by bitfag** [comment]
> Can you add the environment.yml changes to all of the .yml files and not only the linux one?

@carlolm it's a bit problematic as I don't have mac or windows readily available. To update win/mac environments, a one can use the following instruction:

1. Create new temp conda environment `conda create --name tempenv nomkl numpy pandas numexpr`
2. Run `conda env export --name tempenv` to get yml-formatted dependencies
3. Edit setup/environment.yml to remove all mkl stuff and adjust numpy/numexpr versions, same as in 5e115c9532886947179b892c16900c2c692d5a36

**Comment by carlolm** [comment]
> > Can you add the environment.yml changes to all of the .yml files and not only the linux one?
> 
> @carlolm it's a bit problematic as I don't have mac or windows readily available. To update win/mac environments, a one can use the following instruction:
> 
> 1. Create new temp conda environment `conda create --name tempenv nomkl numpy pandas numexpr`
> 2. Run `conda env export --name tempenv` to get yml-formatted dependencies
> 3. Edit setup/environment.yml to remove all mkl stuff and adjust numpy/numexpr versions, same as in [5e115c9](https://github.com/CoinAlpha/hummingbot/commit/5e115c9532886947179b892c16900c2c692d5a36)

Thanks, actually you're right. Since this PR is specific to the dockerfile, we can add the environment.yml fixes for MacOS and Win separately.  Just need to note to try to make them unified, for consistency.

File: setup/environment-linux.yml [hunk 1]
```diff
@@ -22,16 +22,13 @@ dependencies:
   - libstdcxx-ng=9.2.0=hdf63c60_2
   - lz4-c=1.8.1.2=h14c3975_0
   - lzo=2.10=h49e0be7_2
-  - mkl=2020.0=166
-  - mkl-service=2.3.0=py38he904b0f_0
-  - mkl_fft=1.0.15=py38ha843d7b_0
-  - mkl_random=1.1.0=py38h962f231_0
+  - nomkl=3.0=0
```
Comment (carlolm) [inline setup/environment-linux.yml:25]: minor comment, to put this in alphabetical order @bitfag

**Comment by mifeng** [comment]
@dennisocana @bitfag @martinkou @Nullably @carlolm As discussed during US standup let's get this PR merged into the v0.29.0 release

**Comment by carlolm** [comment]
```
# This PR
coinalpha/hummingbot   pr-2013               1ff37c2a5169        4 minutes ago       1.77GB

# Previous build
coinalpha/hummingbot   remove-build-folder   2c5fabe1b745        3 days ago          3.62GB
```

@bitfag I am getting this.  After I enter my HB password in the CLI, it is asking me for a sudo password to hummingbot:

image

**Comment by bitfag** [comment]
Ouch, that was a sudoers config not going into final build stage. Should work now.

**Comment by carlolm** [comment]
I had this error:

```
Traceback (most recent call last):
  File "bin/hummingbot_quickstart.py", line 154, in <module>
    main()
  File "bin/hummingbot_quickstart.py", line 150, in main
    asyncio.get_event_loop().run_until_complete(quick_start(args))
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "bin/hummingbot_quickstart.py", line 90, in quick_start
    hb = HummingbotApplication.main_application()
  File "/home/hummingbot/hummingbot/client/hummingbot_application.py", line 87, in main_application
    cls._main_app = HummingbotApplication()
  File "/home/hummingbot/hummingbot/client/hummingbot_application.py", line 94, in __init__
    input_handler=self._handle_command, bindings=load_key_bindings(self), completer=load_completer(self)
  File "/home/hummingbot/hummingbot/client/ui/completer.py", line 190, in load_completer
    return HummingbotCompleter(hummingbot_application)
  File "/home/hummingbot/hummingbot/client/ui/completer.py", line 37, in __init__
    self._py_file_completer = WordCompleter(file_name_list(SCRIPTS_PATH, "py"))
  File "/home/hummingbot/hummingbot/client/ui/completer.py", line 24, in file_name_list
    return sorted([f for f in listdir(path) if isfile(join(path, f)) and f.endswith(file_extension)])
FileNotFoundError: [Errno 2] No such file or directory: 'scripts/'
```

@bitfag 
I think need to add this line:
```
COPY --chown=hummingbot:hummingbot scripts/ scripts/
```

It works when I added that:
image

**Comment by bitfag** [comment]
@carlolm as I understand scripts supposed to be customized by the user and we don't want to loose them on upgrading. So, a separate docker volume should be used (same as for conf/, data/ etc). As there are defaults scripts, a volume could be pre-populated with these scripts.

**Comment by bitfag** [comment]
One more: a volume will be pre-populated with files only if no `--mount` command is supplied to docker run. So if to create an instance with `installation/docker-commands/create.sh` script, a `hummingbot_files/hummingbot_scripts` directory will still be empty. Though this should be probably addressed separately as it does nothing with image size.

**Comment by carlolm** [comment]
> One more: a volume will be pre-populated with files only if no `--mount` command is supplied to docker run. So if to create an instance with `installation/docker-commands/create.sh` script, a `hummingbot_files/hummingbot_scripts` directory will still be empty. Though this should be probably addressed separately as it does nothing with image size.

Thanks, yes agreed.  This is separate functionality unrelated to this PR.

**Pull request merged**